src/TeamDataView.elm

```diff
@@ -1,24 +1,33 @@
 module TeamDataView exposing (Model, Msg, init, subscriptions, update, view)
 
-import Html exposing (button, div, input, label, text)
-import Html.Attributes as Attributes exposing (placeholder, style, type_, value)
+import Html exposing (button, div, fieldset, input, label, text)
+import Html.Attributes as Attributes exposing (checked, name, placeholder, style, type_, value)
 import Html.Events as Events exposing (onClick, onInput)
 import Http
 import Json.Decode
 import String
 
 
+type Station
+    = Red1 String
+    | Red2 String
+    | Red3 String
+    | Blue1 String
+    | Blue2 String
+    | Blue3 String
+
+
 type Msg
     = ScouterInput String
     | TeamInput String
     | MatchInput String
-    | DriverStationPositionInput String
+    | DriverStationPositionInput String String
```

Comment:
If you use strings, you can hold invalid state.
How about:
```elm
type Color
    = Red
    | Blue

{- Don't ever use `Position` directly, because it allows invalid positions (i.e. `Position -3`, `Position 20`). -}
type Position
    = Position Int


makePosition : Int -> Maybe Position
makePosition n =
    if n > 0 && n < 4 then
        Just <| Position n

    else
        Nothing


type Msg
    ...
    | PickStation Color Position
```
This way, you have only the valid combinations, and you can translate them to any language you choose (just like `elm/time`'s `Month`).

Reply:
Ideally `Position` & `makePosition` should be put in another module so that you really can't directly use `Position` (because only `makePosition` and `Position` should be exposed (not `Position(..)`).